lib/libutil++/libutil++

```diff
@@ -0,0 +1,240 @@
+/*-
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2025 Chelsio Communications, Inc.
+ * Written by: John Baldwin <jhb@FreeBSD.org>
+ */
+
+#ifndef __LIBUTILPP__
+#define	__LIBUTILPP__
+
+#include <sys/cdefs.h>
+#include <sys/nv.h>
+#include <libutil.h>
+#include <netdb.h>
+#include <stdarg.h>
+#include <unistd.h>
+
+#include <memory>
+
+namespace freebsd {
+	/*
+	 * FILE_up is a std::unique_ptr<> for FILE objects which uses
+	 * fclose() to destroy the wrapped pointer.
+	 */
+	struct fclose_deleter {
+		void operator() (FILE *fp) const
+		{
+			fclose(fp);
+		}
+	};
+
+	typedef std::unique_ptr<FILE, fclose_deleter> FILE_up;
+
+	/*
+	 * addrinfo_up is a std::unique_ptr<> which uses
+	 * freeaddrinfo() to destroy the wrapped pointer.  It is
+	 * intended to wrap arrays allocated by getaddrinfo().
+	 */
+	struct freeaddrinfo_deleter {
+		void operator() (struct addrinfo *ai) const
```

Comment:
Style(9) would eat the space after the first ). But This style is more consistent with C++ programs more generally, so we should leave it.